packages/api/cms-api/src/dam/files/dto/file.input.ts

```diff
@@ -89,6 +102,16 @@ export class CreateFileInput {
     // TODO is this validation even used?
     @IsObject()
     scope?: DamScopeInterface;
+
+    @IsString()
+    @IsNotEmpty()
+    @ValidateIf((input) => input.importSourceType !== undefined)
+    importSourceId?: string;
+
+    @IsString()
+    @IsUndefinable()
```

Comment:
I forgot: If we want to either allow both or none we need to set it to

```suggestion
    @IsNotEmpty()
```